chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1339 @@
+
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+        rf.valid_cross_modes = ["Tone->Tone", "Tone->DTCS", "DTCS->Tone",
+                                "->Tone", "->DTCS", "DTCS->", "DTCS->DTCS"]
+
+        rf.valid_characters = chirp_common.CHARSET_ASCII
+        rf.valid_modes = ["FM", "NFM", "AM"]
+        rf.valid_tmodes = ["", "Tone", "TSQL", "DTCS", "Cross"]
+
+        rf.valid_skips = [""]
+
+        # This radio supports memories 1-200, 201-214 are the VFO memories
+        rf.memory_bounds = (1, 214)
+
+        # This is what the BK4819 chip supports
+        # Will leave it in a comment, might be useful someday
+        # rf.valid_bands = [(18000000,  620000000),
+        #                  (840000000, 1300000000)
+        #                  ]
+        rf.valid_bands = []
+        for a in BANDS:
+            rf.valid_bands.append(
+                    (int(BANDS[a][0]*1000000), int(BANDS[a][1]*1000000)))
+        return rf
+
+    # Do a download of the radio from the serial port
+    def sync_in(self):
+        self._mmap = do_download(self)
+        self.process_mmap()
+
+    # Do an upload of the radio to the serial port
+    def sync_out(self):
+        do_upload(self)
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.channel[number-1])
+
+    def validate_memory(self, mem):
+        msgs = super().validate_memory(mem)
+        return msgs
+
+    def _set_tone(self, mem, _mem):
+        ((txmode, txtone, txpol),
+         (rxmode, rxtone, rxpol)) = chirp_common.split_tone_encode(mem)
+
+        if txmode == "Tone":
+            txtoval = CTCSS_TONES.index(txtone)
+            txmoval = 0b01
+        elif txmode == "DTCS":
+            txmoval = txpol == "R" and 0b11 or 0b10
+            txtoval = DTCS_CODES.index(txtone)
+        else:
+            txmoval = 0
+            txtoval = 0
+
+        if rxmode == "Tone":
+            rxtoval = CTCSS_TONES.index(rxtone)
+            rxmoval = 0b01
+        elif rxmode == "DTCS":
+            rxmoval = rxpol == "R" and 0b11 or 0b10
+            rxtoval = DTCS_CODES.index(rxtone)
+        else:
+            rxmoval = 0
+            rxtoval = 0
+
+        _mem.code_flag = (_mem.code_flag & 0b11001100) | (
+            txmoval << 4) | rxmoval
+        _mem.rxcode = rxtoval
+        _mem.txcode = txtoval
+
+    def _get_tone(self, mem, _mem):
+        rxtype = _mem.code_flag & 0x03
+        txtype = (_mem.code_flag >> 4) & 0x03
+        rx_tmode = TMODES[rxtype]
+        tx_tmode = TMODES[txtype]
+
+        rx_tone = tx_tone = None
+
+        if tx_tmode == "Tone":
+            if _mem.txcode < len(CTCSS_TONES):
+                tx_tone = CTCSS_TONES[_mem.txcode]
```

Comment:
The pythonic way to do this is to just do the index into the list and catch the IndexError:
```
try:
    tx_tone = CTCSS_TONES[_mem.txcode]
except IndexError:
    tx_tone = 0